WooCommerce/Classes/ViewRelated/Dashboard/ProductStock/ProductStockDashboardCardViewModel.swift

```diff
@@ -69,9 +77,114 @@ private extension ProductStockDashboardCardViewModel {
+                                                              earliestDateToInclude: currentDate,
+                                                              latestDateToInclude: last30Days,
+                                                              pageSize: Constants.maxItemCount,
+                                                              pageNumber: Constants.pageNumber,
+                                                              orderBy: .itemsSold,
+                                                              order: .descending) { result in
+                continuation.resume(with: result)
+            })
+        }
+    }
+
+    @MainActor
+    func fetchVariationReports(productIDs: [Int64], variationIDs: [Int64]) async throws -> [ProductReport] {
+        let timeZone = TimeZone.siteTimezone
+        let currentDate = Date().endOfDay(timezone: timeZone)
+        let last30Days = Date(timeInterval: -Constants.dayInSeconds*30, since: currentDate).startOfDay(timezone: timeZone)
+        return try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(ProductAction.fetchVariationReports(siteID: siteID,
+                                                                productIDs: productIDs,
+                                                                variationIDs: variationIDs,
+                                                                timeZone: timeZone,
+                                                                earliestDateToInclude: currentDate,
+                                                                latestDateToInclude: last30Days,
+                                                                pageSize: Constants.maxItemCount,
+                                                                pageNumber: Constants.pageNumber,
+                                                                orderBy: .itemsSold,
+                                                                order: .descending) { result in
+                continuation.resume(with: result)
+            })
+        }
+    }
+
+    @MainActor
+    func fetchAndSaveReportsToMemory(for stock: [ProductStock]) async throws {
+
+        let groupedStockByVariations = Dictionary(grouping: stock, by: { $0.isProductVariation })
+        let variationsToFetchReports = groupedStockByVariations[true] ?? []
+        let productsToFetchReports = groupedStockByVariations[false] ?? []
+
+        try await withThrowingTaskGroup(of: Void.self) { [weak self] group in
+            guard let self else { return }
+
+            if variationsToFetchReports.isNotEmpty {
+                group.addTask {
+                    let reports = try await self.fetchVariationReports(
+                        productIDs: Array(Set(variationsToFetchReports.map { $0.parentID })),
+                        variationIDs: variationsToFetchReports.map { $0.productID }
```

Comment:
❓I believe that variation ID is not unique across products. Please correct me if I am wrong about this. 

Considering that the above is true, when we are fetching reports for more than one variation, there could be a conflict in variation IDs. I wonder how the endpoint is able to differentiate variations with the same IDs but they actually belong to different product IDs. 🤔

Reply:
I confirmed with the team in p1716794405419699-slack-C03L1NF1EA3 that variation IDs are globally unique so this solution should work as expected.